sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -206,90 +287,70 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) {
       recoverFromCheckpointLocation: Boolean = true,
       trigger: Trigger = ProcessingTime(0),
       triggerClock: Clock = new SystemClock()): StreamingQuery = {
-    activeQueriesLock.synchronized {
-      val name = userSpecifiedName match {
-        case Some(n) =>
-          if (activeQueries.values.exists(_.name == userSpecifiedName.get)) {
+    // `queryName` and `queryId` will be set when we add them into `pendingQueryNames` and
+    // `pendingQueryIds` so that we can remove them at the end of this method when an error happens.
+    var queryName: String = null
+    var queryId: UUID = null
+    try {
+      userSpecifiedName.foreach { name =>
```

Comment:
Would be nice to add comment in each section, like "Make sure no other query with same name is active", "...query with same id...." , etc.